src/main/flight/position_control.c

```diff
@@ -48,3 +58,50 @@ const pidCoefficient_t *getAltitudePidCoeffs(void)
 {
     return &altitudePidCoeffs;
 }
+
+void resetAltitudeControl (void) {
+    altitudeI = 0.0f;
+}
+
+void altitudeControl(float targetAltitudeCm, float taskIntervalS, float verticalVelocity, float targetAltitudeStep) {
+
+    float altitudeErrorCm = targetAltitudeCm - getAltitudeCm();
+    const float altitudeP = altitudeErrorCm * altitudePidCoeffs.Kp;
+
+    // reduce the iTerm gain for errors greater than 2m, otherwise it winds up too much
+    const float largeError = 200.0f; // 2m
+    const float itermRelax = (fabsf(altitudeErrorCm) < largeError) ? 1.0f : 0.1f;
+    altitudeI += altitudeErrorCm * altitudePidCoeffs.Ki * itermRelax * taskIntervalS;
+    const float iTermLimit = 200.0f;
+    altitudeI = constrainf(altitudeI, -iTermLimit, iTermLimit); 
+
+    const float altitudeD = verticalVelocity * altitudePidCoeffs.Kd;
+
+    const float altitudeF = targetAltitudeStep * altitudePidCoeffs.Kf;
+
+    const float throttleAdjustment = altitudeP + altitudeI - altitudeD + altitudeF;
+    const float tiltMultiplier = 2.0f - fmaxf(getCosTiltAngle(), 0.5f);
+    // 1 = flat, 1.24 at 40 degrees, max 1.5 around 60 degrees, the default limit of Angle Mode
+    // 2 - cos(x) is between 1/cos(x) and 1/sqrt(cos(x)) in this range
+
+    const float hoverThrottle = positionControlConfig()->hover_throttle - PWM_RANGE_MIN;
+
+    float newThrottle = (hoverThrottle + throttleAdjustment) * tiltMultiplier + PWM_RANGE_MIN;
```

Comment:
`PWM_RANGE_MIN` (=750) is important for analog signals. I don't quite understand what `PWM_RANGE_MIN` has to do with the minimum throttle value. Isn't this supposed to be 1000, not 750?

Reply:
There are a number of potential issues relating to throttle.  This is how it has been done in GPS rescue for several years now.
I agree that the throttle adjustment calculations should be reviewed, but that is something for a separate PR.
All I did here was re-use the existing code from GPS Rescue.
This PR is intended to be refactoring only.
We can look into the best solution for throttle calculation in a separate PR.